asb-ballerina/commons.bal

```diff
@@ -46,12 +38,77 @@ public enum receiveModes {
 #                      Endpoint=sb://namespace_DNS_Name;EntityPath=EVENT_HUB_NAME;
 #                      SharedAccessKeyName=SHARED_ACCESS_KEY_NAME;SharedAccessKey=SHARED_ACCESS_KEY or  
 #                      Endpoint=sb://namespace_DNS_Name;EntityPath=EVENT_HUB_NAME;
-#                      SharedAccessSignatureToken=SHARED_ACCESS_SIGNATURE_TOKEN
-@display {label: "Connection Config"}
-public type AsbConnectionConfiguration record {|
-    @display {label: "Connection String"}
+#                      SharedAccessSignatureToken=SHARED_ACCESS_SIGNATURE_TOKEN 
+# + entityConfig -  This field holds the configuration details of either a topic or a queue. The type of the entity is 
+#                   determined by the entityType field. The actual configuration details are stored in either a 
+#                   TopicSubsConfig or a QueueConfig record  
+# + entityType - This field is an enumeration (EntityType) that indicates whether the entity is a topic or a queue  
+# + receiveMode - This field holds the receive modes(RECEIVE_AND_DELETE/PEEK_LOCK) for the connection. The receive mode determines how messages are 
+#                 retrieved from the entity. The default value is PEEK_LOCK
+@display {label: "Receiver Connection Config"}
+public type ASBServiceReceiverConfig record {
+    @display {label: "Entity Type"}
+    EntityType entityType;
+    @display {label: "Entity Configuration"}
+    TopicSubsConfig|QueueConfig entityConfig;
+    @display {label: "ConnectionString"}
+    string connectionString;
```

Comment:
I know fields in a record has no order. But for config panel rendering purposes what we make order as

    @display {label: "ConnectionString"}
    string connectionString;
    @display {label: "Entity Type"}
    EntityType entityType;
    @display {label: "Entity Configuration"}
    TopicSubsConfig|QueueConfig entityConfig;

However, do we need `EntityType ` as it is captured by entityConfig record's type?